homeassistant/components/zhong_hong/climate.py

```diff
@@ -2,138 +2,65 @@
 
 from __future__ import annotations
 
+import asyncio
 import logging
 from typing import Any
 
-import voluptuous as vol
-from zhong_hong_hvac.hub import ZhongHongGateway
-from zhong_hong_hvac.hvac import HVAC as ZhongHongHVAC
-
 from homeassistant.components.climate import (
     ATTR_HVAC_MODE,
-    FAN_HIGH,
-    FAN_LOW,
-    FAN_MEDIUM,
-    FAN_MIDDLE,
-    PLATFORM_SCHEMA as CLIMATE_PLATFORM_SCHEMA,
     ClimateEntity,
     ClimateEntityFeature,
     HVACMode,
 )
-from homeassistant.const import (
-    ATTR_TEMPERATURE,
-    CONF_HOST,
-    CONF_PORT,
-    EVENT_HOMEASSISTANT_STOP,
-    UnitOfTemperature,
-)
+from homeassistant.const import ATTR_TEMPERATURE, UnitOfTemperature
 from homeassistant.core import HomeAssistant
-from homeassistant.helpers import config_validation as cv
-from homeassistant.helpers.dispatcher import (
-    async_dispatcher_connect,
-    async_dispatcher_send,
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import ZhongHongConfigEntry
+from .const import (
+    DOMAIN,
+    MANUFACTURER,
+    MODEL,
+    ZHONG_HONG_MODE_COOL,
+    ZHONG_HONG_MODE_DRY,
+    ZHONG_HONG_MODE_FAN_ONLY,
+    ZHONG_HONG_MODE_HEAT,
 )
-from homeassistant.helpers.entity_platform import AddEntitiesCallback
-from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+from .coordinator import ZhongHongDataUpdateCoordinator
 
 _LOGGER = logging.getLogger(__name__)
 
-CONF_GATEWAY_ADDRRESS = "gateway_address"
-
-DEFAULT_PORT = 9999
-DEFAULT_GATEWAY_ADDRRESS = 1
-
-SIGNAL_DEVICE_ADDED = "zhong_hong_device_added"
-SIGNAL_ZHONG_HONG_HUB_START = "zhong_hong_hub_start"
-
-PLATFORM_SCHEMA = CLIMATE_PLATFORM_SCHEMA.extend(
-    {
-        vol.Required(CONF_HOST): cv.string,
-        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
-        vol.Optional(
-            CONF_GATEWAY_ADDRRESS, default=DEFAULT_GATEWAY_ADDRRESS
-        ): cv.positive_int,
-    }
-)
```

Comment:
We should try to import the old configuration so users only have to remove the YAML configuration